jit-compiler/src/compiler.rs

```diff
@@ -60,6 +59,35 @@ pub fn generate_glue_code<T: FieldElement>(
     Ok(format!("{PREAMBLE}\n{glue}\n",))
 }
 
+const PREAMBLE: &str = r#"
+#![allow(unused_parens)]
+
+#[derive(Clone, Copy)]
+struct FieldElement(u64);
```

Comment:
Why do we define it from scratch, instead of using the implementation we already have? Would it take too long to compile? Is there a plan to support larger fields?

Reply:
There are several possibilities, but I first wanted to check if JIT compilation is feasible at all. With the crate we currently use, it takes sooo much longer. Also, for the JIT it would be sufficient to use a crate that only supports one specific field, while our current number create supports all fields that we currently have.

The first milestone is being able to evaluate array expressions efficiently. For that, we do not need arithmetic on field elements.